Feat/token interaction

sumitnitsurat opened the pull request:
- Displaying token owner and asset info in detailed verify block.
- Refactored Detailed verify block component
- Added token verify block component

<img width="1155" alt="Screenshot 2020-01-03 at 11 43 17" src="https://user-images.githubusercontent.com/12955874/71706186-4208f000-2e1e-11ea-89be-7805d0d0a1cf.png">

Nebulis commented on src/components/home/dropZone/views/defaultView.test.js:
```diff
@@ -0,0 +1,24 @@
+import React from "react";
```
I'm not fond of using snapshot for such testing:
- `expect(tree).toMatchSnapshot();` doesn't give any hints on what we expect, i.e what is important in the generated html
- same apply between two tests, what's the difference in term of expectation ? Needs to compare both snapshot and check the diff
- over the time we need to always update the snapshot for any html change (even if it doesnt break anything

What do you think if we just do expectation on what has been rendered ? my assumption is that depending `hover` and `accept` values some part of the component are displayed => let's directly make assumption on this

sumitnitsurat committed:
Added token owner information from oa-tone

sumitnitsurat committed:
test added for token owner detail view

sumitnitsurat committed:
color changed for transferable asset

sumitnitsurat committed:
fixed integration failed test

sumitnitsurat committed:
variable name fixed

sumitnitsurat committed:
variable name fixed

sumitnitsurat committed:
test added for detailed certioficate verify block

sumitnitsurat committed:
integration test fixed

sumitnitsurat committed:
oa-token updated

sumitnitsurat committed:
test updated with shallow

sumitnitsurat committed:
tets for certificateVerifyBlock added

rjchow commented on src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/TokenVerifyBlock.js:
```diff
@@ -0,0 +1,31 @@
+  return (
+    <>
+      <div className={css["transferable-record"]}>The document is a transferable record.</div>
+      <div>Owned by: {tokenOwner}</div>
```
use makeEtherscanURL here so that its clickable

sumitnitsurat replied:
added

rjchow commented on src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/TokenVerifyBlock.js:
```diff
@@ -0,0 +1,31 @@
+        const owner = await getTokenOwner();
+        setTokenOwner(owner);
+      } catch (e) {
+        setError(e.message);
```
do a debug error call here too so that we can easily debug this in the future if there's something wrong with the rendering

sumitnitsurat replied:
added.

rjchow commented on src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/TokenVerifyBlock.js:
```diff
@@ -0,0 +1,31 @@
+    async function fetchTokenOwner() {
+      try {
+        await initializeToken(document);
+        const owner = await getTokenOwner();
```
this call takes awhile so i think we should put a spinner here - on my side it takes awhile to show up then there's a reflow after that which is a bit annoying

rjchow replied:
alternatively move this call up to the main page level which might make more sense - because we're passing in the document all the way in here, which to me seems like a contradiction to the existing design which only receives statuses as props

sumitnitsurat replied:
moved up to parent. need to add loader/spinner because its still taking time. created a new issue for that. As its the UI improvement.

rjchow commented on src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/DetaledCertificateVerifyBlock.test.js:
```diff
@@ -0,0 +1,31 @@
+import React from "react";
```
this file's filename spelling

sumitnitsurat replied:
fixed.

rjchow commented on src/services/token/token.js:
```diff
@@ -0,0 +1,20 @@
+import { ReadOnlyToken, Writeable } from "@govtechsg/oa-token";
```
wonder why this didn't fail, the exported member name is "WriteableToken" - yea the docs is wrong

sumitnitsurat replied:
wasn`t using that. fixed it though.

rjchow commented on src/components/CertificateVerifyBlock/DetailedCertificateVerifyBlock/CertificateVerifyCheck.js:
```diff
@@ -1,8 +1,7 @@
 import { get } from "lodash";
-import css from "./detailedCertificateBlock.scss";
-import { TYPES, MESSAGES } from "../../constants/VerificationErrorMessages";
+import { TYPES, MESSAGES } from "../../../constants/VerificationErrorMessages";
```
i think sin yong mentioned something about changing all 'certificate' to 'document' or something? do you remember precisely?

sumitnitsurat replied:
yea..use document instead of certificate. Will change in another PR. create a new issue for that.

rjchow commented on src/services/token/token.js:
```diff
@@ -0,0 +1,20 @@
+
+export const initializeToken = async (document, web3Provider = undefined, wallet = undefined) => {
+  tokenInstance = await (web3Provider && wallet
+    ? new Writeable({ document, web3Provider, wallet })
```
add logs for the branches here so that we can debug this easily

sumitnitsurat replied:
added.

sumitnitsurat committed:
Merge branch 'master' into feat/token-interaction

sumitnitsurat committed:
PR changes

- Etherscan link added
- Filename fixed
- Moved data fetching to parent component

sumitnitsurat committed:
test added for fetching owner error

sumitnitsurat committed:
Merge branch 'master' into feat/token-interaction

sumitnitsurat committed:
build fixed
